Add VarPasing options.

geonmo opened the pull request:
Add VarPasing options for 72X runCat.py

jhgoh commented on create-batch:
```diff
@@ -178,6 +180,16 @@ class TheJobConfig():
+        if options.useMiniAOD :
+          additional_options += "useMiniAOD="+str(options.useMiniAOD)+" "
+        if options.globalTag :
+          additional_options += "globalTag="+str(options.globalTag) 
```
I think this is just for CATTools.
Could you make it more general way? Wouldn't it be working with sys.argv for example?

jhgoh replied:
My proposal for the use case, 
create-batch ... --args="runOnMC=blah_blah_blah globalTag=foo_bar" ...

Then self.additional_options = opts["--args"]

geonmo commented:
I changed to dedicated to general option.
However, this has a limitation about option's ordering.
For example, 
```bash
create-batch -n --jobName cat_prod --fileList filelist.txt --maxFiles 3 --args "runOnMC=True useMiniAOD=True globalTag='PHYS14_25_V2::All'" --cfg runCat.py
```
is working correctly.
However,
```bash
create-batch -n --jobName cat_prod --fileList filelist.txt --maxFiles 3  --cfg runCat.py --args "runOnMC=True useMiniAOD=True globalTag='PHYS14_25_V2::All'"
```
is not working when it do VarParsiong function

=Log
```bash
[geonmo@ui10 prod]$ create-batch -n --jobName cat_prod --fileList filelist.txt --maxFiles 3 --cfg runCat.py --args "runOnMC=True useMiniAOD=True globalTag='PHYS14_25_V2::All'"
@@ Preparing batch jobs in cat_prod ...
@@ Loading python cfg...
Traceback (most recent call last):
  File "/cms/home/geonmo/hep-tools/create-batch", line 464, in <module>
    jobConfig.initialiseWorkspace()
  File "/cms/home/geonmo/hep-tools/create-batch", line 183, in initialiseWorkspace
    process = __import__('.'.join(self.cfgFileName.split('.')[:-1])).process
  File "./runCat.py", line 12, in <module>
    options.parseArguments()
  File "/cvmfs/cms.cern.ch/slc6_amd64_gcc481/cms/cmssw-patch/CMSSW_7_2_2_patch1/python/FWCore/ParameterSet/VarParsing.py", line 245, in parseArguments
    raise RuntimeError, "Unknown command"
RuntimeError: Unknown command
```

jhgoh commented on create-batch:
```diff
@@ -178,6 +181,7 @@ class TheJobConfig():
         cout = sys.stdout
         sys.stdout = open("%s/log.txt" % self.jobDir, "w")
         process = __import__('.'.join(self.cfgFileName.split('.')[:-1])).process
+        options = __import__('.'.join(self.cfgFileName.split('.')[:-1])).options
```
This line can make usual cmssw cfg to be crashed. We cannot guarantee "options".

geonmo commented:
Hello, I removed that line.
Regards,

geonmo commented:
sys.argv will be clear when create-batch try to parse options.
Run command) 
```bash
create-batch --jobName aodtest --fileList filelist_AODDY.txt --maxFiles 1 --args "globalTag=PHYS14_25_V2::All runOnMC=True useMiniAOD=False" --cfg runCat_cfg.py
```

jhgoh commented on create-batch:
```diff
@@ -177,6 +180,10 @@ class TheJobConfig():
         cout = sys.stdout
         sys.stdout = open("%s/log.txt" % self.jobDir, "w")
+        sys.argv=[]
+        sys.argv.append(".py")
```
Is this a dummy argument?

geonmo replied:
".py" is required by "VarParsing module" at python.
VarParsing module check ".py" to delimination options.

jhgoh replied:
Thank you. I merged it.

jhgoh commented on create-batch:
```diff
@@ -177,6 +180,10 @@ class TheJobConfig():
+        sys.argv=[]
+        sys.argv.append(".py")
+        sys.argv.extend( self.additional_options.split())
+        print sys.argv
```
We can remove this line.

jhgoh commented on create-batch:
```diff
@@ -253,7 +260,7 @@ else
     echo TERMINATED_$EXITCODE `date` cmsRun job_${{SECTION}}_cfg.py >> {0}/submit.log
     exit 1
-fi""".format(self.jobDir, self.jobBase, os.environ['CMS_PATH'])
+fi""".format(self.jobDir, self.jobBase, os.environ['CMS_PATH'],self.additional_options)
```
For safety when the additional_options are empty, I think it may be better to append empty string.
``if len(self.additional_options) == 0: self.additional_options = ""``
